apps/beeai-ui/src/app/(auth)/utils.ts

```diff
@@ -27,54 +35,79 @@ export async function jwtWithRefresh(
     // Subsequent logins, but the `access_token` has expired, try to refresh it
     if (!token.refresh_token) throw new TypeError('Missing refresh_token');
 
+    const tokenProvider = providers.find(({ name }) => name === token.provider);
+    if (!tokenProvider) {
+      throw new TypeError('No matching provider found');
+    }
+
+    // Type assertion to ensure we have the OIDC options
+    const providerOptions = tokenProvider.options as OIDCProviderOptions | undefined;
+
+    if (!providerOptions?.clientId || !providerOptions?.clientSecret || !providerOptions?.issuer) {
+      throw new TypeError('Missing clientId, clientSecret, or issuer in provider configuration');
+    }
+
+    const { clientId, clientSecret, issuer: issuerUrl } = providerOptions;
+
+    let refreshTokenUrl: string;
+
     try {
-      // The `token_endpoint` can be found in the provider's documentation. Or if they support OIDC,
-      // at their `/.well-known/openid-configuration` endpoint.
-      // lookup the provider's uri
-      const tmp = providers.filter((p) => p.name === token['provider']);
-      if (tmp.length === 0) {
-        throw new TypeError('no matching provider found');
-      }
-      const tokenProvider = tmp[0];
+      // Try to discover the OIDC configuration and get the token endpoint
+      const config = await openidClient.discovery(new URL(issuerUrl), clientId, clientSecret);
```

Comment:
I don't think there is caching in the library, this is too slow to do repeatedly. Doing it once on the server boot should be fine, the data doesn't change very often.